### PR TITLE
feat(CX-3785): update my collection sticky tab logic

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3785] 

### Description
Update my collection sticky tab logic

https://github.com/artsy/eigen/assets/11945712/4bc9a10e-fd93-435b-9dee-b8752af7c6db

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Update sticky parts logic inside my collection

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3785]: https://artsyproduct.atlassian.net/browse/CX-3785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 1]
```diff
@@ -3,15 +3,21 @@ import {
...
+  Text,
   Touchable,
+  TouchableHighlightColor,
+  bullet,
   useSpace,
 } from "@artsy/palette-mobile"
 import AsyncStorage from "@react-native-async-storage/async-storage"
 import { ArtworksFilterHeader } from "app/Components/ArtworkGrids/ArtworksFilterHeader"
-import SearchIcon from "app/Components/Icons/SearchIcon"
+// import { Input } from "app/Components/Input"
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:18]: 👀

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 2]
```diff
@@ -106,7 +112,8 @@ export const MainStickyHeader: React.FC<{ hasArtworks: boolean }> = ({ hasArtwor
         alignItems="center"
         flexDirection="row"
         justifyContent="space-between"
-        style={{ paddingTop: space(2), paddingHorizontal: space(2) }}
+        py={2}
+        style={{ paddingHorizontal: space(2) }}
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:116]: 👀 
```suggestion
        p={2}
```

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 3]
```diff
@@ -292,13 +300,69 @@ const Filters = ({ filtersCount, showModal }: { filtersCount: number; showModal:
...
+  return (
+    <Flex backgroundColor="white100" flexDirection="row" pb={1}>
+      <Input
+        testID="MyCollectionSearchBarInput"
+        placeholder="Search Your Artworks"
+        onChangeText={setQuery}
+        enableClearButton
+        value={query}
+        returnKeyType="done"
+        // fontSize={16}
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:327]: 👀

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 4]
```diff
@@ -292,13 +300,69 @@ const Filters = ({ filtersCount, showModal }: { filtersCount: number; showModal:
...
+        autoCorrect={false}
+        style={{
+          marginLeft: space(2),
+        }}
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:331]:
```suggestion
        ml={2}
```

**Review by MrSltun — APPROVED** [review]
Looks good! 🌟  I just left a few comments

**Review by dariakoko — APPROVED** [review]

**Comment by MounirDhahri** [comment]
@gkartalis 


Screenshot 2023-06-14 at 14 12 32
Screenshot 2023-06-14 at 14 07 19

**Review by gkartalis — COMMENTED** [review]
noice! Thanks!

**Pull request merged**